### DISCUSSION TITLE
read lots values at once

### DISCUSSION THREAD

**Discussion by maccardo** [pymodbus-dev/pymodbus#2195] [General]
I was wondering if the device I am reading contain different values accessed by registers, how can I read them once as a block so I only establish IO overhead once and not for every value, and then decode them as a block. Does such a feature exist, how do other do this or is this something that is not done in general.

**Comment by janiversen** [comment]
The modbus protocol support read_registers but only if they are in consecutive addresses.

Be aware that most devices do not support "a lot", especially not on a serial link.

Please take a look at the modbus protocol standard, we provide a link in the documentation
